compiler/condsyms.nim

```diff
@@ -166,4 +166,5 @@ proc initDefines*(symbols: StringTableRef) =
   defineSymbol("nimHasWarnStdPrefix")
 
   defineSymbol("nimHasVtables")
+  defineSymbol("nimHasGenericsOpenSym")
```

Comment:
shouldn't this be defined only when the experimental is on?

Reply:
The nimHas defines usually signal that the experimental switch is available, checking if it's enabled is trickier since it can change depending on the context with `push experimental` etc. #23933 was opened recently related to this

Reply:
turns out `nimHasGenericsOpenSym` is already declared, but for the broken version shipped in 2.0.8:

- https://github.com/nim-lang/Nim/blob/v2.0.8/compiler/condsyms.nim#L103
- https://github.com/nim-lang/Nim/blob/v2.0.8/compiler/options.nim#L224

will need a new name for this new, working version :tada:

Reply:
`nimHasGenericsOpenSym2`?

Reply:
Changed to `nimHasGenericsOpenSym2`